weaviate/collections/batch/base.py

```diff
@@ -186,12 +185,55 @@ def __init__(
         self.__last_scale_up: float = 0
         self.__max_observed_rate: int = 0
 
-        self.__loop = event_loop
-
+        self.__loop: Optional[asyncio.AbstractEventLoop] = self.__start_new_event_loop()
         self.__start_bg_thread()
 
+    def __run_event_loop(self, loop: asyncio.AbstractEventLoop) -> None:
+        loop.set_debug(True)  # in case of errors, shows async errors in the thread to users
+        asyncio.set_event_loop(loop)
+        try:
+            loop.run_forever()
+        finally:
+            # This is entered when loop.stop is scheduled from the main thread
+            loop.run_until_complete(loop.shutdown_asyncgens())
+            loop.close()
+
+    def __start_new_event_loop(self) -> asyncio.AbstractEventLoop:
+        self.__loop = asyncio.new_event_loop()
```

Comment:
you are setting it here and also return it, where it is also set - maybe have it as a local variable here?